user-module/controller/user-controller.js

```diff
@@ -171,5 +172,129 @@ class UserController {
       next(error)
     }
   }
+
+  async getUser (req, res) {
+    try {
+      logger.info('Getting user by Email')
+      const { username } = req.body
+      const data = await userService.getUser(username)
+      logger.info(defaultMessage.SUCCESS)
+      responseService.onSuccess(res, defaultMessage.SUCCESS, data, defaultStatusCode.SUCCESS)
+    } catch (error) {
+      logger.error(error, defaultMessage.ERROR)
+      responseService.onError(res, defaultMessage.ERROR, error)
+    }
+  }
+
+  async updateUser (req, res) {
+    try {
+      logger.info('Updating user')
+      const user = req.body
+      const userInfo = await userService.getUser(user.username)
+      if (userInfo !== null) {
+        const data = await userService.updateUser(user)
+        logger.info(defaultMessage.SUCCESS)
+        responseService.onSuccess(res, defaultMessage.SUCCESS, data, defaultStatusCode.SUCCESS)
+      } else {
+        logger.error(defaultMessage.ERROR)
+        responseService.onError(res, defaultMessage.ERROR)
+      }
+    } catch (error) {
+      logger.error(error, defaultMessage.ERROR)
+      responseService.onError(res, defaultMessage.ERROR, error)
+    }
+  }
+
+  async getUserList (req, res) {
+    try {
+      logger.info('Getting user list')
+      const data = await userService.getUserList()
+      logger.info(defaultMessage.SUCCESS)
+      responseService.onSuccess(res, defaultMessage.SUCCESS, data, defaultStatusCode.SUCCESS)
+    } catch (error) {
+      logger.error(error, defaultMessage.ERROR)
+      responseService.onError(res, defaultMessage.ERROR, error)
+    }
+  }
+
+  async addRole (req, res) {
+    try {
+      logger.info('Adding new role')
+      if (req.body.role !== null && req.body.role !== '') {
```

Comment:
use object destruction to get role form the req body.
null check can be done in the following format
`if (!role)` Please use this syntax to null check else where